nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/diagnostics/bootstrap/shell/ShellCommandExecutor.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.diagnostics.bootstrap.shell;
+
+import org.apache.nifi.diagnostics.bootstrap.shell.commands.AbstractShellCommand;
+
+import java.io.IOException;
+import java.util.Collection;
+
+public class ShellCommandExecutor {
+
+    public static Collection<String> execute(final AbstractShellCommand command) {
```

Comment:
Given the open-ended nature of this utility class, recommend moving it to the `commands` package and removing the `public` visibility.

Reply:
ShellCommandExecutor is called from OperatingSystemDiagnosticTask which is in the tasks package. Removing the public visibility would make in inaccessible for this class.

Reply:
Thanks, moving the method sounds like a good approach.